packages/transform/src/template-to-typescript.ts

```diff
@@ -77,16 +79,39 @@ export function templateToTypescript(
       }
     }
 
-    function emitTemplateBoilerplate(emitBody: () => void): void {
-      emit.text(`({} as typeof import("${typesPath}")).template(function`);
-      emit.synthetic(typeParams);
-      emit.text('(𝚪');
+    function emitTemplateBoilerplate(emitBody: () => void, useJsDoc: boolean): void {
+      if (useJsDoc) {
+        emit.text(
+          `${
+            contextType ? '/** @type {unknown} */ (' : ''
+          }/** @type {import("${typesPath}")} */ ({}).template(function(`
+        );
```

Comment:
[nit] Could this be broken into two `emit.text` calls? This took me a couple of tries to read 😄 

```suggestion
        if (contextType) {
          emit.text('/** @type {unknown} */ (');
        }
        emit.text(`(/** @type {typeof import("${typesPath}")} */ ({})).template(function(`);
```